doc/Language/compilation.pod6

```diff
@@ -0,0 +1,295 @@
+The directory names are the ASCII encoded SHA-1 hashes of the UTF-8 encoded module short-names.
+
+In these directories we find one file per distribution that contains a module with a matching short name.
+These files again contain the ID of the dist and the other fields that make up the long name: auth, version and api.
+So by reading these files we have a usually short list of auth-version-api triplets which we can match against our
+DependencySpecification.
+We end up with the winning dist's ID, which we use to look up the meta data, stored in a JSON encoded file.
+This meta data contains the name of the file in the sources/ directory containing the requested module's code.
+This is what we can load.
+
+Finding names for source files is again a bit tricky, as there's still the Unicode issue and in addition the same
+relative file names may be used by different installed distributions (think versions).
+So for now at least, we use SHA-1 hashes of the long-names.
+
+=head2 Resources
+
+=begin code
+%?RESOURCES
+%?RESOURCES<libraries/p5helper>
+%?RESOURCES<icons/foo.png>
+%?RESOURCES<schema.sql>
+
+Foo
+|___ lib
+|     |____ Foo.rakumod
+|
+|___ resources
+      |___ schema.sql
+      |
+      |___ libraries
+            |____ p5helper
+            |        |___
+            |___ icons
+                     |___ foo.png
+
+=end code
+
+It's not only source files that are stored and found this way.
+Distributions may also contain arbitrary resource files.
+These could be images, language files or shared libraries that are compiled on installation.
+They can be accessed from within the module through the C<%?RESOURCES> hash
+
+As long as you stick to the standard layout conventions for distributions, this even works during development
+without installing anything.
+
+A nice result of this architecture is that it's fairly easy to create special purpose repositories.
+
+=head2 Dependencies
+
+Luckily precompilation at least works quite well in most cases. Yet it comes with its own set of challenges.
+Loading a single module is easy.
+The fun starts when a module has dependencies and those dependencies have again dependencies of their own.
+
+When loading a precompiled file in C<Raku> we need to load the precompiled files of all its dependencies, too.
+And those dependencies B<must> be precompiled, we cannot load them from source files.
+Even worse, the precomp files of the dependencies B<must> be exactly the same files we used for precompiling our
+module in the first place.
+
+To top it off, precompiled files work only with the exact C<Raku> binary, that was used for compilation.
+
+All of that would still be quite manageable if it weren't for an additional requirement: as a user you expect a new
+version of a module you just installed to be actually used, don't you?
+
+In other words: if you upgrade a dependency of a precompiled module, we have to detect this and precompile the module
+again with the new dependency.
+
+=head2 Precomp stores
```

Comment:
I don't think this tutorial actually achieves what it was slated to do. It's a nice overview of the whole piñata, but I'd rather see it documented piecewise where it would be more useful, and just leave a small, actionable piece of advice.
The problem is that it explains a few concepts, but as such it belongs to the reference manual, not the tutorial. Tutorials must explain how to do; in this case, I thought that the main intention was to explain how to use precompilation programmatically. Maybe that's not the case.